### DISCUSSION TITLE
Unreliably entering raw REPL mode (possibly due to reset?)

### DISCUSSION THREAD

**Discussion by pn2new** [micropython/micropython#16268] [ESP32]
I use `pyboard.py` to transfer files to my board, either using the command line or importing Pyboard from my own python script. In both cases, scripts fail regularly to enter raw REPL mode. I tried using soft_reset or not, but this does not solve the problem.

The `data` that is printed before the exception is raised, is this:

`b'ets Jun  8 2016 00:22:57\r\n\r\nrst:0x10 (RTCWDT_RTC_RESET),boot:0x17 (SPI_FAST_FLASH_BOOT)\r\nconfigsip: 0, SPIWP:0xee\r\nclk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00\r\nmode:DIO, clock div:2\r\nload:0x3fff0030,len:4892\r\nho 0 tail 12 room 4\r\nload:0x40078000,len:14896\r\nload:0x40080400,len:4\r\nload:0x40080404,len:3372\r\nentry 0x400805b0\r\nMicroPython v1.24.0 on 2024-10-25; Generic ESP32 module with ESP32\r\nType "help()" for more information.\r\n>>> '
` 
This is the message that is printed after a reset (I can trigger this with `machine.reset()`).

What could be the reason for this behaviour?

**Comment by shariltumin** [comment]
Have you tried ```mpremote```?

https://docs.micropython.org/en/latest/reference/mpremote.html

**Comment by bidrohini** [comment]
You can adjust the watchdog timer settings in your MicroPython code:
import machine
machine.WDT(timeout=0)  # Disable the WDT
